src/overlays/ovl_i6/fox_sy.c

```diff
@@ -307,108 +280,3034 @@ void func_i6_801983E4(Boss* boss) {
+                gDPSetEnvColor(gMasterDisp++, 32, 255, 32, 255);
+                Matrix_Pop(&gGfxMatrix);
+                Matrix_Push(&gGfxMatrix);
+                Matrix_Translate(gGfxMatrix, boss->fwork[38], boss->fwork[39], boss->fwork[40], 1);
+                Matrix_RotateY(gGfxMatrix, boss->obj.rot.y * M_DTOR, 1);
+                Matrix_RotateX(gGfxMatrix, boss->obj.rot.x * M_DTOR, 1);
+                Matrix_RotateZ(gGfxMatrix, boss->obj.rot.z * M_DTOR, 1);
+                Matrix_Scale(gGfxMatrix, boss->fwork[42] * 2.0f, boss->fwork[42], boss->fwork[42] * 2.0f, 1);
+                Matrix_SetGfxMtx(&gMasterDisp);
+                gSPDisplayList(gMasterDisp++, D_1024AC0);
+            }
+        }
+        if (boss->fwork[43] != 0.0f) {
+            RCP_SetupDL_49();
+            gDPSetPrimColor(gMasterDisp++, 0x00, 0x00, 0, 0, 0, 255);
+            gDPSetEnvColor(gMasterDisp++, 0, 0, 0, 0);
+            Matrix_Pop(&gGfxMatrix);
+            Matrix_Push(&gGfxMatrix);
+            Matrix_Translate(gGfxMatrix, boss->obj.pos.x, 157.0f, boss->obj.pos.z, 1);
+            Matrix_RotateX(gGfxMatrix, (M_PI / 2), 1);
+            Matrix_Scale(gGfxMatrix, boss->fwork[43], boss->fwork[43], boss->fwork[43], 1);
+            Matrix_SetGfxMtx(&gMasterDisp);
+            gSPDisplayList(gMasterDisp++, D_1024AC0);
+        }
+        if (boss->fwork[46] != 0.0f) {
+            RCP_SetupDL_49();
+            gDPSetPrimColor(gMasterDisp++, 0x00, 0x00, (s32) boss->fwork[46], (s32) boss->fwork[46], 0,
+                            (s32) boss->fwork[46]);
+            gDPSetEnvColor(gMasterDisp++, 0, 0, 0, 0);
+            Matrix_Pop(&gGfxMatrix);
+            Matrix_Push(&gGfxMatrix);
+            Matrix_Translate(gGfxMatrix, boss->obj.pos.x + 10.0f, boss->obj.pos.y + 70.0f, boss->obj.pos.z + 60.0f, 1);
+            Matrix_Scale(gGfxMatrix, 0.4f, 0.2f, 0.2f, 1);
+            Matrix_SetGfxMtx(&gMasterDisp);
+            gSPDisplayList(gMasterDisp++, D_1024AC0);
+            Matrix_Translate(gGfxMatrix, -46.0f, 0, 0, 1);
+            Matrix_SetGfxMtx(&gMasterDisp);
+            gSPDisplayList(gMasterDisp++, D_1024AC0);
+        }
+    }
+}
+
+static Vec3f D_i6_801A69FC[10] = {
+    { 0.0f, -80.0f, 800.0f },      { -500.0f, -100.0f, 400.0f },  { 500.0f, -140.0f, 400.0f },
+    { 3250.0f, 600.0f, 3500.0f },  { -2000.0f, 400.0f, 5000.0f }, { 2500.0f, 400.0f, 5000.0f },
+    { -1000.0f, 400.0f, 6000.0f }, { 250.0f, 400.0f, 2500.0f },   { -2000.0f, 0.0f, 4000.0f },
+    { 2500.0f, 0.0f, 4000.0f },
+};
+
+void func_i6_8019EB80(void) {
+    Actor* actor;
+    s32 i;
+    s32 var_s1;
+
+    actor = gActors;
+    if (gHitCount >= 100) {
+        var_s1 = 9;
+    } else {
+        var_s1 = 2;
+    }
+    Rand_SetSeed(1, 0x71AC, 0x263A);
+
+    for (i = 0; i <= (var_s1); i++, actor++) {
+        if (((i != 0) || !(gTeamShields[3] <= 0.0f)) && ((i != 1) || !(gTeamShields[2] <= 0.0f)) &&
+            ((i != 2) || !(gTeamShields[1] <= 0.0f))) {
```

Comment:
```suggestion
        if (((i == 0) && (gTeamShields[3] <= 0.0f)) || ((i = 1) && (gTeamShields[2] <= 0.0f))||
            ((i == 2) || (gTeamShields[1] <= 0.0f))) {
            continue;
            }
```
This gets rid of the negated float compares.